### PR TITLE
8317510: Change Windows debug symbol files naming to avoid losing info when an executable and a library share the same name

### PR DESCRIPTION
**Pull request by fthevenet**
When building OpenJDK on Windows using "--with-native-debug-info=external", the resulting debug symbols are saved in files located in the same folder as the corresponding executable or library and named by swapping the extension ".exe" or ".dll" for a ".pdb" one (or "diz" if option "--with-native-debug-info=zipped" is used), which means that in the event of an exe and a dll file sharing the same target folder and file name (e.g. `bin\java.exe` and `bin\java.dll`), we have to choose whether symbols in `bin\java.pdb` will refer to the exe or the dll; we can't have both.

This PR addresses this issue by adopting a different naming strategy for the resulting symbol files where we keep the full name of every file - including its `dll` or `exe` extension) and then add the appropriate `.pdb`, `.map` or `.diz` extension .

For instance,  `jvm.dll` symbols are no longer called `jvm.pdb` but instead `jvm.dll.pdb`. Similarly, it is now `jvm.dll.diz` when using zipped symbols, and "jvm.dll.stripped.pdb" for stripped symbols (i.e. when "--with-external-symbols-in-bundles=public" is used).

The PR also removes the existing filtering for java.pdb, jimage.pdb and jpackage.pdb used to guaranty the dll symbols were bundled over the ones from the exe, since we no longer need that.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8317510](https://bugs.openjdk.org/browse/JDK-8317510): Change Windows debug symbol files naming to avoid losing info when an executable and a library share the same name (**Enhancement** - P4)

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16039/head:pull/16039` \
`$ git checkout pull/16039`

Update a local copy of the PR: \
`$ git checkout pull/16039` \
`$ git pull https://git.openjdk.org/jdk.git pull/16039/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16039`

View PR using the GUI difftool: \
`$ git pr show -t 16039`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16039.diff">https://git.openjdk.org/jdk/pull/16039.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16039#issuecomment-1747305751)

### CODE REVIEW

**Comment by fthevenet** [comment]
The following build scenarios where tested on Windows, and the symbols name manually verified for correctness:
* `./configure --with-native-debug-symbols=external`
* `./configure --with-native-debug-symbols=zipped`
* `./configure --with-native-debug-symbols=external --with-external-symbols-in-bundles=full`
* `./configure --with-native-debug-symbols=external --with-external-symbols-in-bundles=public`

For sanity check, I also ran the following on Linux (where we're not expecting to see any differences):
* `./configure --with-native-debug-symbols=external`
* `./configure --with-native-debug-symbols=zipped`
* `./configure --with-native-debug-symbols=external --with-external-symbols-in-bundles=full`

**Comment by erikj79** [comment]
I tried to build this internally and hit this error:

```
make[3]: *** No rule to make target '/cygdrive/c/sb/prod/1696445346/workspace/build/windows-x64/hotspot/variant-server/libjvm/gtest/jvm.pdb', needed by '/cygdrive/c/sb/prod/1696445346/workspace/build/windows-x64/images/test/hotspot/gtest/server/jvm.pdb'.  Stop.
```

Looks like the gtest build is failing.

File: make/CreateJmods.gmk [hunk 1]
```diff
@@ -81,13 +81,11 @@ endif
 ifneq ($(CMDS_DIR), )
   DEPS += $(call FindFiles, $(CMDS_DIR))
   ifeq ($(call isTargetOs, windows)+$(SHIP_DEBUG_SYMBOLS), true+public)
```
Comment (TheShermanTanker) [inline make/CreateJmods.gmk:83]:
```suggestion
  ifeq ($(call isCompiler, microsoft)+$(SHIP_DEBUG_SYMBOLS), true+public)
```

stripped pdbs (and pdbs in general) are only ever used with the VS toolchain

File: make/CreateJmods.gmk [hunk 2]
```diff
@@ -81,13 +81,11 @@ endif
...
-    # For public debug symbols on Windows, we have to use stripped pdbs, rename them
-    # and filter out a few launcher pdbs where there's a lib that goes by the same name
+    # For public debug symbols on Windows, we have to use stripped pdbs and rename them
```
Comment (TheShermanTanker) [inline make/CreateJmods.gmk:84]:
```suggestion
    # For public debug symbols on VS, we have to use stripped pdbs and rename them
```

File: make/CreateJmods.gmk (hunk 1 above)
Comment (fthevenet) [reply make/CreateJmods.gmk:83]: I understand the point of this change, but it is not directly with the issue addressed here (i.e. this condition wasn't introduced in this PR.).
Should it be included in the PR anyway?

Comment (TheShermanTanker) [reply make/CreateJmods.gmk:83]: On second thought, hold on for a while

**Comment by fthevenet** [comment]
I have added a basic test that verifies that symbols can be resolved by the internal jdk tooling that makes use of them, even after the name change for the .pdb files.
It relies on the fact that on Windows/MSVC, native frames contains no function names when dumped in hs_err if pdb are not available/cannot be loaded.
This is not true for all platforms (e.g. on Linux, function names are present in hs_err even when full symbols aren't available), so I have restricted this test to run on Windows only.

File: make/CreateJmods.gmk (hunk 1 above)
Comment (erikj79) [reply make/CreateJmods.gmk:83]: I agree with @fthevenet, such a change should be separate from this.

**Comment by erikj79** [comment]
> I have added a basic test that verifies that symbols can be resolved by the internal jdk tooling that makes use of them, even after the name change for the .pdb files. It relies on the fact that on Windows/MSVC, native frames contains no function names when dumped in hs_err if pdb are not available/cannot be loaded. This is not true for all platforms (e.g. on Linux, function names are present in hs_err even when full symbols aren't available), so I have restricted this test to run on Windows only.

Adding a test seems nice, but I will have to defer to Hotspot people for reviewing the validity, placement and tier/group inclusion of the new test. My interpretation is that with the current placement and TEST.groups it will be part of tier1.

**Comment by fthevenet** [comment]
Yes, of course.
Please note that, many other tests in the same group, it will not be run in non-debug builds; this is because it relies on `-XX:ErrorHandlerTest` to cause a controlled crash of the JVM (so that the hs_err file is produced) and this is only available in debug builds (fast or slow).
I _think_ this precludes this test from being part of tiers 1 (which is what I intended for), but I may very well be wrong.

**Comment by fthevenet** [comment]
OK, so unsurprisingly I was indeed wrong... The test ran as part or tier1 in the GHA checks - and failed.
I'll let other advise on whether or not running this is tier1  is appropriate or not, but in the meantime I'll try to figure out why it fails in GHA.

**Comment by tstuefe** [comment]
> Yes, of course. Please note that, like many other tests in the same group, it will not be run in non-debug builds; this is because it relies on `-XX:ErrorHandlerTest` to cause a controlled crash of the JVM (so that the hs_err file is produced) and this is only available in debug builds (fast or slow). I _think_ this precludes this test from being part of tiers 1 (which is what I intended for), but I may very well be wrong.

They will run in tier1. 

Test looks good.

Cheers, Thomas

**Comment by tstuefe** [comment]
> OK, so unsurprisingly I was indeed wrong... The test ran as part or tier1 in the GHA checks - and failed. I'll let other advise on whether or not running this is tier1 is appropriate or not, but in the meantime I'll try to figure out why it fails in GHA.

Looks like symbol resolution of the crash address failed:
```
TEST RESULT: Failed. Execution failed: `main' threw exception: java.lang.RuntimeException: '# V \[jvm.dll.*\].*(crash_with_segfault|controlled_crash).*' missing from stdout/stderr
```

looking at the output of the crashing process, it just prints out a PC, but not the expected frame:
```
# Problematic frame:
# V  [jvm.dll+0xfdd069]
```
where we expected to see a function name.

**Comment by fthevenet** [comment]
According to the GHA workflow, tier1 test are run with a jdk build using the following `configure` line, with doesn't set `--with-native-debug-symbols`:
```
bash configure --with-conf-name=windows-x64 --with-debug-level=fastdebug --with-version-opt=${GITHUB_ACTOR}-${GITHUB_SHA} --with-boot-jdk=/d/a/jdk/jdk/bootjdk/jdk --with-jtreg=jtreg/installed --with-gtest=gtest --with-msvc-toolset-version=14.29 --with-jmod-compress=zip-1 
```
The documentation isn't much help in advising what the default is in this case, but if it equivalent to  `--with-native-debug-symbols:none` , then this explains why the test fails.
I'll dig a bit deeper to confirm that.

**Comment by erikj79** [comment]
> > I have added a basic test that verifies that symbols can be resolved by the internal jdk tooling that makes use of them, even after the name change for the .pdb files. It relies on the fact that on Windows/MSVC, native frames contains no function names when dumped in hs_err if pdb are not available/cannot be loaded. This is not true for all platforms (e.g. on Linux, function names are present in hs_err even when full symbols aren't available), so I have restricted this test to run on Windows only.
> 
> Adding a test seems nice, but I will have to defer to Hotspot people for reviewing the validity, placement and tier/group inclusion of the new test. My interpretation is that with the current placement and TEST.groups it will be part of tier1.

Verified that the test passes in our internal build and test environment (which runs hotspot tests in tier1 on debug builds).

**Comment by erikj79** [comment]
> According to the GHA workflow, tier1 test are run with a jdk build using the following `configure` line, which doesn't set `--with-native-debug-symbols`:
> 
> ```
> bash configure --with-conf-name=windows-x64 --with-debug-level=fastdebug --with-version-opt=${GITHUB_ACTOR}-${GITHUB_SHA} --with-boot-jdk=/d/a/jdk/jdk/bootjdk/jdk --with-jtreg=jtreg/installed --with-gtest=gtest --with-msvc-toolset-version=14.29 --with-jmod-compress=zip-1 
> ```
> 
> The documentation isn't much help in advising what the default is in this case, but if it is equivalent to `--with-native-debug-symbols:none` , then this explains why the test fails. I'll dig a bit deeper to confirm that.

I would expect the default to be to include pdbs on Windows (`external`).

**Comment by TheShermanTanker** [comment]
The default for all platforms is none on a static build, and external otherwise
https://github.com/openjdk/jdk/blob/053f45695f9a914452f39029bd9ac7eb329e6883/make/autoconf/jdk-options.m4#L262

File: make/CreateJmods.gmk (hunk 1 above)
Comment (TheShermanTanker) [reply make/CreateJmods.gmk:83]: I'll retract my suggestion then, and include this change after this is integrated on my own

**Comment by TheShermanTanker** [comment]
P.S. I suggest changing the comment in the new test that states "where the function names are available even with no symbols". This is not very accurate, the reason it works is because the symbols are inside libjvm.so itself, unlike with Microsoft Visual C which places all debug information (including symbol names) into the pdb files entirely. (For instance, if you compile the Windows Java VM with gcc, the symbol names would also be inside the VM itself, and not in the debug files)

**Comment by fthevenet** [comment]
From what I can tell, the test fails because of an unrelated issue with the "test-prebuilt" target used to run the test by GHA.

According to the logs for the failed test, path to the symbol folder "/d/a/jdk/jdk/bundles/symbols/jdk-22/fastdebug" is passed explicitly via the env variable `SYMBOLS_IMAGE_DIR`: 
```
make test-prebuilt TEST='test/hotspot/jtreg/:tier1_runtime' BOOT_JDK=/d/a/jdk/jdk/bootjdk/jdk JT_HOME=jtreg/installed JDK_IMAGE_DIR=/d/a/jdk/jdk/bundles/jdk/jdk-22/fastdebug SYMBOLS_IMAGE_DIR=/d/a/jdk/jdk/bundles/symbols/jdk-22/fastdebug TEST_IMAGE_DIR=/d/a/jdk/jdk/bundles/tests JTREG='JAVA_OPTIONS=-XX:-CreateCoredumpOnCrash;VERBOSE=fail,error,time;KEYWORDS=!headful'
```

But looking into the hs_err report, we see the path for the symbols folder is not considered by the symbol engine:
```
symbol engine: initialized successfully - sym options: 0x614 - pdb path: .;D:\a\jdk\jdk\bundles\jdk\jdk-22\fastdebug\bin;C:\Windows\SYSTEM32;C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.17763.4851_none_de72d1b65349cfc4;D:\a\jdk\jdk\bundles\jdk\jdk-22\fastdebug\bin\server
```

That would still imply that the pdb files were stripped from the jdk bundle that is uploaded by the build task, or they would be picked by the symbol engine from there. Right now, I don't know if this is indeed the case, and if so caused by the renaming of pdb files in this PR.

**Comment by erikj79** [comment]
> From what I can tell, the test fails because of an unrelated issue with the "test-prebuilt" target used to run the test by GHA.
> 
> According to the logs for the failed test, path to the symbol folder "/d/a/jdk/jdk/bundles/symbols/jdk-22/fastdebug" is passed explicitly via the env variable `SYMBOLS_IMAGE_DIR`:
> 
> ```
> make test-prebuilt TEST='test/hotspot/jtreg/:tier1_runtime' BOOT_JDK=/d/a/jdk/jdk/bootjdk/jdk JT_HOME=jtreg/installed JDK_IMAGE_DIR=/d/a/jdk/jdk/bundles/jdk/jdk-22/fastdebug SYMBOLS_IMAGE_DIR=/d/a/jdk/jdk/bundles/symbols/jdk-22/fastdebug TEST_IMAGE_DIR=/d/a/jdk/jdk/bundles/tests JTREG='JAVA_OPTIONS=-XX:-CreateCoredumpOnCrash;VERBOSE=fail,error,time;KEYWORDS=!headful'
> ```
> 
> But looking into the hs_err report, we see the path for the symbols folder is not considered by the symbol engine:
> 
> ```
> symbol engine: initialized successfully - sym options: 0x614 - pdb path: .;D:\a\jdk\jdk\bundles\jdk\jdk-22\fastdebug\bin;C:\Windows\SYSTEM32;C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.17763.4851_none_de72d1b65349cfc4;D:\a\jdk\jdk\bundles\jdk\jdk-22\fastdebug\bin\server
> ```
> 
> That would still imply that the pdb files were stripped from the jdk bundle that is uploaded by the build task, or they would be picked by the symbol engine from there. Right now, I don't know if this is indeed the case, and if so caused by the renaming of pdb files in this PR.

I must confess that I'm not very well versed in the github actions configuration and execution model, nor our implementation. In our internal build-and-test system, we provide `SYMBOLS_IMAGE_DIR` as an env variable when invoking `test-prebuilt`, and the test passes in this setup. From your description it seems our GHA are supposed to be setup in the same way, but maybe there is a bug in there somewhere.

**Comment by fthevenet** [comment]
> In our internal build-and-test system, we provide SYMBOLS_IMAGE_DIR as an env variable when invoking test-prebuilt, and the test passes in this setup. From your description it seems our GHA are supposed to be setup in the same way, but maybe there is a bug in there somewhere.

If it isn't too much of a bother, could you please verify that the path you pass via SYMBOLS_IMAGE_DIR is indeed listed in the symbol engine search path in the resulting hs_err produced as part of the test?
I have been doing some testing locally (without GHA) and I'm seeing weird things in the symbol engine logs in hs_err, like the path passed to SYMBOLS_IMAGE_DIR not being converted to Windows path, while JDK_IMAGE_DIR is, e.g.:

Command line:
`make test-prebuilt BOOT_JDK=/cygdrive/c/java-21-openjdk-21.0.0.0.35-7.win.jdk.x86_64/ JT_HOME=../../upstream/jtreg/build/images/jtreg JDK_IMAGE_DIR=build/prebuilt_images/jdk/ SYMBOLS_IMAGE_DIR=build/prebuilt_images/symbols/ TEST_IMAGE_DIR=build/prebuilt_images/test/ TEST="test/hotspot/jtreg/runtime/errorhandling/TestSymbolsInHsErrFile.java"`

hs_err:
`symbol engine: initialized successfully - sym options: 0x614 - pdb path: .; build/prebuilt_images/symbols//bin:build/prebuilt_images/symbols//bin/server;C:\openjdk\jdk\build\prebuilt_images\jdk\bin;C:\Windows\SYSTEM32;C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.19041.1110_none_60b5254171f9507e;C:\\openjdk\jdk\build\prebuilt_images\jdk\bin\server`

Also please note that if pdb files are present in the folder that you pass via JDK_IMAGE_DIR, the symbol engine will likely pick them up from there even if the path set in SYMBOLS_IMAGE_DIR is empty or incorrect (and therefore the test will pass).

**Comment by fthevenet** [comment]
All that to say that it looks like there might be a bug in ` test-prebuilt` (which doesn't mean there couldn't be one GHA too, of course...)

Apologies if I'm just shooting in the dark here, but at a cursory glance, I notice that SYMBOLS_IMAGE_DIR isn't checked for validity like JDK_IMAGE_DIR or TEST_IMAGE_DIR are; do you know why that is? Although from what I understand, it wouldn't transform the path, it would have caught the problem, wouldn't it?

https://github.com/openjdk/jdk/blob/80232b7e753129ca7a4f1ca9b70844e0c7d8eabf/make/RunTestsPrebuilt.gmk#L121C9-L124

**Comment by erikj79** [comment]
> All that to say that it looks like there might be a bug in ` test-prebuilt` (which doesn't mean there couldn't be one GHA too, of course...)
> 
> Apologies if I'm just shooting in the dark here, but at a cursory glance, I notice that in RunTestsPrebuilt.gmk, SYMBOLS_IMAGE_DIR isn't checked for validity like JDK_IMAGE_DIR or TEST_IMAGE_DIR are; do you know why that is? Although from what I understand, it wouldn't transform the path, it would have caught the problem, wouldn't it?
> 
> https://github.com/openjdk/jdk/blob/80232b7e753129ca7a4f1ca9b70844e0c7d8eabf/make/RunTestsPrebuilt.gmk#L121C9-L124

We aren't checking SYMBOLS_IMAGE_DIR because we consider it optional. Tests are expected to be able to run without it, you just get less information on crashes.

_NT_SYMBOL_PATH is setup with Windows style paths based on SYMBOLS_IMAGE_DIR here: https://github.com/openjdk/jdk/blob/80232b7e753129ca7a4f1ca9b70844e0c7d8eabf/make/RunTests.gmk#L63

In our run-test-prebuilt scenario, our JDK_IMAGE_DIR does not have any pdb files. The _NT_SYMBOLS_PATH setup was introduced to provide pdb files for symbol lookup in hs_err files, so I'm fairly certain that part works.

**Comment by fthevenet** [comment]
I see, thanks!
I'll try to figure out why the symbol path isn't converted to Windows path in my own local tests.

**Comment by magicus** [comment]
Something looks definitely fishy in our `get-bundles` action in GHA. It seems that the symbol path is not correctly exported to `test.yml`. It has this "how can this ever has worked???" look. I'll dig more into it...

**Comment by magicus** [comment]
Debugging GHA is a royal pain. It seems reasonable that the pdb file is generated correctly by the build:
```
Copying images/symbols/bin/server/jvm.dll.pdb
```

The symbols bundle seem to be correctly uploaded after the build.
```
Uploaded D:\a\jdk\jdk\bundles\symbols-windows-x64.tar.gz (100.0%) bytes 41943040:44545171
```

The bundle seems to be properly downloaded and unzipped  by the test script:
```
Unpacking symbols bundle...
```

The symbols dir sent to RunTest looks sane, but I can't say if it is really is correct:
```
SYMBOLS_IMAGE_DIR=/d/a/jdk/jdk/bundles/symbols/jdk-22/fastdebug
```

To verify this, you could add a patch like this:
```
--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -148,6 +148,10 @@ jobs:
             echo "value=$PATH" >> $GITHUB_OUTPUT
           fi
 
+      - name: 'Check symbols'
+        run: >
+          ls -R ${{ steps.bundles.outputs.symbols-path }}
+
       - name: 'Run tests'
         id: run-tests
         run: >
```

**Comment by magicus** [comment]
You could also instrument RunTest.gmk to show if `_NT_SYMBOL_PATH` is actually set properly.

**Review by magicus — APPROVED** [review]
The code itself looks good. We just need to figure out why it is breaking on GHA.

**Comment by TheShermanTanker** [comment]
Once integrated, this is going to spam everyone on GitHub with Windows test failures until the issue is fixed separately :/

**Comment by tstuefe** [comment]
> Once integrated, this is going to spam everyone on GitHub with Windows test failures until the issue is fixed separately :/ (Could we perhaps split the test out into another change?)

I don't think @fthevenet plans on checking in a broken test. He will figure this out or disable the test.

**Comment by fthevenet** [comment]
Indeed, I see no reason to rush the integration for this before we've resolved the failing test.

I can confirm that outside of the context of GHA, the `test-prebuilt` target does not properly resolve _relative_ paths that are passed to SYMBOLS_IMAGE_DIR, but works fine with absolute path.
This is therefore likely *not* the reason for the test failing in GHA, since this uses absolute paths (i.e. `/d/a/jdk/jdk/bundles/symbols/jdk-22/fastdebug`). 
This is still annoying, as the other paths used as parameters by the same `test-prebuilt` do accept related paths, but beside the point here, so I'll leave it that for now and maybe come back to it in separately.

Meanwhile, I have instrumented the GHA test workflow as @magicus suggested and waiting for the results.

**Comment by magicus** [comment]
@fthevenet Did the instrumentation give anything?

**Comment by fthevenet** [comment]
@magicus As a matter of fact, it did yield some interesting resutls, but then I got quite caught up with the release earlier this week and this and didn't find the time to update this thread.

The listing of the folder that is passed as a parameter to the test run shows it contains the pdb files with the expected names (e.g. jvm.dll.pdb). You can see an example of that in the "Check symbols" step in the run below:
https://github.com/fthevenet/jdk/actions/runs/6480669248/job/17599856238

I also added a print out of the values for SYMBOL_PATH and _NT_SYMBOL_PATH when they are first assigned (see https://github.com/fthevenet/jdk/commit/5655ca70b20767401905a86ac41be8035b686dff#diff-041bf69ea79b333b9ce99c1f879e398d698538530a35c361500b72631f059233R70), but to my surprise I could not see those in the test run logs from GHA, while are indeed printed when I run the test locally.

One notable difference I noticed, is that I run all my local tests using cygwin, while GHA uses MSYS2; could this explain anything?

**Comment by magicus** [comment]
> One notable difference I noticed, is that I run all my local tests using cygwin, while GHA uses MSYS2; could this explain anything?

Yes, msys2/cygwin are sufficiently different that it could possibly explain this.

**Comment by fthevenet** [comment]
The root cause for this test failing turns out to be indeed related to MSYS2 ; namely checks made to determine whether or not we're running on Windows return "false", so all Windows specific code is ignored.
In this case, we simply jump over the part where _NT_SYMBOL_PATH is set.

I opened https://bugs.openjdk.org/browse/JDK-8318669 to make the logic that auto detects the target OS when running pre-built test  in "RunTestsPrebuilt.gmk" aware of MSYS2, but I am now wondering if it might not be best to just set OPENJDK_TARGET_OS explicitly to  "windows" in the command line that launch the tests?

**Comment by fthevenet** [comment]
I finally opted to address the underlying issue by patching RunTestsPrebuilt.gmk, rather than GHA; https://github.com/openjdk/jdk/pull/16343.

As for this PR, I see two possible ways forward; one is to remove the test and integrate the change without it as part of the current PR, and add the test back in a follow up once the RunTestsPrebuilt patch is integrated.
The other is to convert this PR to a draft, wait for the separate fix to be integrated, and then rebase this PR on top of it and resume its review.

I like the first one better; a few more steps but overall less fussy. I'm also open to another solution.

**Comment by tstuefe** [comment]
> I finally opted to address the underlying issue by patching RunTestsPrebuilt.gmk, rather than GHA; #16343.
> 
> As for this PR, I see two possible ways forward; one is to remove the test and integrate the change without it as part of the current PR, and add the test back in a follow up once the RunTestsPrebuilt patch is integrated. The other is to convert this PR to a draft, wait for the separate fix to be integrated, and then rebase this PR on top of it and resume its review.
> 
> I like the first one better; a few more steps but overall less fussy. I'm also open to another solution.

+1 Vote for first option.

**Comment by magicus** [comment]
You don't have to rebase, in fact, you should not rebase an open PR. Just merge from master, once the msys fix is in.

Your fix for JDK-8318669 was simple; just fix the "else ifeq" as Erik suggests, and you're good to go. So just delay pushing this a bit more -- you don't have to move it to Draft either. I think that is preferable, since it will keep the test with this PR, where it belongs.

**Comment by fthevenet** [comment]
@magicus So, just so that I get this strait, what you're suggesting is that now #16343 is integrated, I merge master _into_ my branch for the PR and push it, right?
(sorry for being a bit dense 😅)

**Comment by erikj79** [comment]
> @magicus So, just so that I get this strait, what you're suggesting is that now #16343 is integrated, I merge master _into_ my branch for the PR and push it, right? (sorry for being a bit dense 😅)

Yes, but perhaps wait for the GHA to finish to verify that it all really works together.

**Comment by fthevenet** [comment]
Test now passes in GHA.

**Review by erikj79 — APPROVED** [review]

**Comment by TheShermanTanker** [comment]
Aw, I wanted to sponsor this one in particular :(

Oh well, it's great that it's integrated now!

**Pull request closed without merging**